### PR TITLE
Custom API Key

### LINKED ISSUE #37: Errors on connection drop
It is an amazing and useful plugin, but when my internet drops I have this stack of errors, I suppose is not a hard problem to fix.
In my opinion, it is better to not see an error and just to switch to the next play.
If someone can help with a fix or with a hint how to fix this,  it will be very nice and useful. 
image

### PR DESCRIPTION
**Pull request by cloewen8**
Allows the user to enter Google Developer Project information to take control of the plugin with their own metrics and quotas.

- The built-in key, client ID and secret will be used if one is not provided.
- If the user provides a key, the built-in OAuth client will not be used, preventing Google security errors.
- The user can specify their own OAuth client ID and secret to login using.
- Fixes a crash with relatedPlaylists.

Any feedback, changes or suggestions are appreciated.

Resolves #37

### CODE REVIEW

File: OptionsDialog.cpp [hunk 1]
```diff
@@ -232,10 +235,11 @@ void OptionsDialog::LoadProfileInfo(std::function<void()> onFinished) {
         if (d.IsObject() && d.HasMember("items") && d["items"].IsArray() && d["items"].Size() > 0 && d["items"][0].HasMember("contentDetails")) {
             const rapidjson::Value &i = d["items"][0]["contentDetails"]["relatedPlaylists"];
 
-            m_userPlaylists.push_back({ Tools::ToWString(i["favorites"]), Plugin::instance()->Lang(L"YouTube.Playlists\\Favorites"), true, L"Favorites" });
-            m_userPlaylists.push_back({ Tools::ToWString(i["uploads"]), Plugin::instance()->Lang(L"YouTube.Playlists\\Uploads"), false, L"Uploads" });
-            m_userPlaylists.push_back({ Tools::ToWString(i["likes"]), Plugin::instance()->Lang(L"YouTube.Playlists\\Likes"), true, L"Likes" });
-            m_userPlaylists.push_back({ Tools::ToWString(i["watchLater"]), Plugin::instance()->Lang(L"YouTube.Playlists\\WatchLater"), true, L"WatchLater" });
+            // YouTube seems to no longer provide favorites or watchLater
```
Comment (AdrianEddy) [inline OptionsDialog.cpp:238]: Hmm, the comment says that `favorites` or `watchLater` may not be present, but the code checks for `favorites` and `uploads`?

Comment (cloewen8) [reply OptionsDialog.cpp:238]: I made them all optional encase Google changes the api further. If there was a common ID I would provide it. There was no common ID for favorites.

Comment (AdrianEddy) [reply OptionsDialog.cpp:238]: but `watchLater` and `likes` are not checked

Comment (cloewen8) [reply OptionsDialog.cpp:238]: For watchLater and likes they have shorthand ids, so I use a ternary there. I'd imagine they should still exist, even if not in relatedPlaylists.

Comment (AdrianEddy) [reply OptionsDialog.cpp:238]: oh right, I see it now

**Comment by AdrianEddy** [comment]
Looks good, thanks!

**Pull request merged**